File: ueueo-framework/ueueo-domain-driven-desgin/src/main/java/com/ueueo/ddd/domain/entities/events/NullEntityChangeEventHelper.cs
namespace Volo.Abp.Domain.Entities.Events;

/**
 * Null-object implementation of <see cref="IEntityChangeEventHelper"/>.
*/
public class NullEntityChangeEventHelper : IEntityChangeEventHelper
{
    /**
     * Gets single instance of <see cref="NullEntityChangeEventHelper"/> class.
    */
    public static NullEntityChangeEventHelper Instance;//  { get; } = new NullEntityChangeEventHelper();

    private NullEntityChangeEventHelper()
    {
    }

    public void PublishEntityCreatingEvent(Object entity)
    {
    }

    public void PublishEntityCreatedEvent(Object entity)
    {
    }

    public void PublishEntityUpdatingEvent(Object entity)
    {
    }

    public void PublishEntityUpdatedEvent(Object entity)
    {
    }

    public void PublishEntityDeletingEvent(Object entity)
    {
    }

    public void PublishEntityDeletedEvent(Object entity)
    {
    }
}

File: ueueo-framework/ueueo-domain-driven-desgin/src/main/java/com/ueueo/ddd/domain/entities/events/NullEntityChangeEventHelper.java
package com.ueueo.ddd.domain.entities.events;

import lombok.Getter;

/**
 * Null-object implementation of <see cref="IEntityChangeEventHelper"/>.
 */
@Getter
public class NullEntityChangeEventHelper implements IEntityChangeEventHelper {
    /**
     * Gets single instance of <see cref="NullEntityChangeEventHelper"/> class.
     */
    private static NullEntityChangeEventHelper Instance = new NullEntityChangeEventHelper();

    private NullEntityChangeEventHelper() {
    }

    @Override
    public void publishEntityCreatedEvent(Object entity) {

    }

    @Override
    public void publishEntityUpdatedEvent(Object entity) {

    }

    @Override
    public void publishEntityDeletedEvent(Object entity) {

    }
}
